src/shared/unit-file.c

```diff
@@ -276,8 +284,8 @@ int unit_file_build_name_map(
 
                         /* search_path is ordered by priority (highest first). If the name is already mapped
                          * to something (incl. itself), it means that we have already seen it, and we should
-                         * ignore it here. */
-                        if (hashmap_contains(ids, de->d_name))
+                         * ignore it here. Except the unit is masked, then try to get the withdrawal path */
+                        if ((revisit_masked = hashmap_contains(ids, de->d_name)) && !set_contains(masked_ids, de->d_name))
```

Comment:
not a fan of such nested assignment expressions. Just use two lines for this, i.e. first "revisitmasked = hashmap_contains(…") and then in a second line the if check

Reply:
done